packages/ui/src/components/Button/index.tsx

```diff
@@ -1,41 +1,104 @@
+import { I18nKey } from '@logto/phrases-ui';
 import classNames from 'classnames';
-import { ReactNode } from 'react';
+import { HTMLProps, ReactElement, ReactNode, useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 
+import DangerousRaw from '../../../../console/src/components/DangerousRaw';
+import { Ring as Spinner } from '../../../../console/src/components/Spinner';
 import * as styles from './index.module.scss';
 
-export type Props = {
+export type ButtonType =
+  | 'primary'
+  | 'secondary'
+  | 'danger'
+  | 'outline'
+  | 'plain'
+  | 'default'
+  | 'branding';
+
+type BaseProps = Omit<HTMLProps<HTMLButtonElement>, 'type' | 'size' | 'title'> & {
   htmlType?: 'button' | 'submit' | 'reset';
+  type?: ButtonType;
+  size?: 'small' | 'medium' | 'large';
   isDisabled?: boolean;
-  className?: string;
-  children: ReactNode | Record<string, unknown>;
-  type?: 'primary' | 'secondary' | 'outline';
-  size?: 'small' | 'large';
-  onClick?: React.MouseEventHandler;
+  isLoading?: boolean;
+  loadingDelay?: number;
```

Comment:
Currenly we do not apply any loading status in the UI button, that is on purpose due to our different design strategies for UI and console.

Reply:
Noted, will remove this.